### ISSUE TITLE
Homebrew crashes when verifying a XCI dump with missing node data in NSWReleases.xml.

### ISSUE THREAD

**Issue by Matt07211** [DarkMatterCore/gcdumptool#14]
Using version v1.0.5
Switch Firmware is the latest availble
SD Card: fat32 (So i use the split option when dumping 
I've tried dumping the game twice now (restarting the deuce the first time to try and rule any random thing out), and the software would crash around the latter half of the dumping (Cant get specifics as I am not paying attention to the device while it dumps)  process.

Logs attached are all that I could find in relation to me dumping the game.
[2-dump-logs.zip](https://github.com/DarkMatterCore/gcdumptool/files/3071804/2-dump-logs.zip)

Note: I am on the latest switch version, which at the time of writing is 7.0.1 and I am using Atmosphere version 0.8.6

Edit: This is also happening for Super Mario Bros U Deluxe

**Comment by DarkMatterCore** [comment]
@Matt07211 

Hi there. Sorry for taking so long to answer, a lot of things have happened in my personal life in the past few months, and they have kept me busy enough to stop browsing the Internet as much as I'd like.

Could you please try out the attached test build? It was built using the latest devkitA64 and libnx releases, and fixes some problems related to fs-srv calls that should likely provide a better stability overall.

[gcdumptool_test_20190422.zip](https://github.com/DarkMatterCore/gcdumptool/files/3104155/gcdumptool_test_20190422.zip)

**Comment by Matt07211** [comment]
Okay, I have tried it with Super Mario Bros u deluxe and it still comes up with this software has crashed due to an error.

**Comment by DarkMatterCore** [comment]
Any error codes?

**Comment by Matt07211** [comment]
Nothing visible on screen (or after clicking okay in the error message). I'd assume the logs will be the same/similar to what is attached in my original message.

**Comment by DarkMatterCore** [comment]
It would be good to have a copy of them as well. Offsets will be different for this build.

**Comment by Matt07211** [comment]
Give me a few hours then I'll be able to grab and upload them

**Comment by DarkMatterCore** [comment]
Thanks! I'll be waiting.

**Comment by Matt07211** [comment]
Sorry for the late response!
Here the logs requested. I attempted it twice so I included both logs.
[dumps.zip](https://github.com/DarkMatterCore/gcdumptool/files/3106743/dumps.zip)

**Comment by DarkMatterCore** [comment]
Sorry to bother you yet again, but could you perform the same test with the v1.0.6 release binary? There were some last minute changes over there.

**Comment by Matt07211** [comment]
Still crashed (Newest Atmosphere and the newest released binary from you. But this time I was sitting ready to take photos. It seems to crash when verifying if the images are to be believed.

Dumping at 99%
![IMG_20190423_213410__01](https://user-images.githubusercontent.com/12839977/56578126-2f43e780-6610-11e9-8c8b-f0bd4649fa91.jpg)

Dumping at 100% and image taken quite literally as the error message was popping up
![IMG_20190423_213412__01](https://user-images.githubusercontent.com/12839977/56578127-2fdc7e00-6610-11e9-837c-c957458046ce.jpg)

Error message
![IMG_20190423_213437](https://user-images.githubusercontent.com/12839977/56578128-2fdc7e00-6610-11e9-8338-82675d63028f.jpg)

**Comment by Matt07211** [comment]
Same thing on dark souls, sorry the pictures arent as good this time around.
![IMG_20190423_215402__01](https://user-images.githubusercontent.com/12839977/56579118-b003e300-6612-11e9-91f0-db6e9d153107.jpg)
![IMG_20190423_215403__01](https://user-images.githubusercontent.com/12839977/56579119-b09c7980-6612-11e9-8621-90924cc0e3a6.jpg)

**Comment by DarkMatterCore** [comment]
Thanks. Now I see where the problem is:

```
<release>
	<id>1187</id>
	<name>New Super Mario Bros. U Deluxe</name>
	<publisher>Nintendo</publisher>
	<region>WLD</region>
	<languages>en,fr,de,it,es,nl,pt,ru,ja,zh,ko</languages>
	<group>HR</group>
	<imagesize>4</imagesize>
	<serial>LA-H-ADALA</serial>
	<titleid>0100EA80032EA000</titleid>
	<imgcrc>72D25E75</imgcrc>
	<filename>hr-adala</filename>
	<releasename>New_Super_Mario_Bros._U_Deluxe_NSW-HR</releasename>
	<trimmedsize>0</trimmedsize>
	<firmware></firmware>
	<type>1</type>
	<card></card>
</release>
```

Notice the missing value for the `<card>` node?

I'll do a quick fix and upload it ASAP for you to test.

**Comment by Matt07211** [comment]
Tell me when, and I'll give it a crack!

**Comment by DarkMatterCore** [comment]
Here's it. Please test the attached build.

[gcdumptool_20190423_xmlfix.zip](https://github.com/DarkMatterCore/gcdumptool/files/3107463/gcdumptool_20190423_xmlfix.zip)

**Comment by DarkMatterCore** [comment]
Coincidentally, Dark Souls shares the same problem as well in the XML:

```
<release>
	<id>1064</id>
	<name>Dark Souls: Remastered</name>
	<publisher>Bandai Namco Entertainment</publisher>
	<region>EUR</region>
	<languages>en,fr,de,it,es,pt,ru,ja</languages>
	<group>LiGHTFORCE</group>
	<imagesize>8</imagesize>
	<serial>LA-H-AK63B</serial>
	<titleid>01004AB00A260000</titleid>
	<imgcrc>D0E40546</imgcrc>
	<filename>lfc-ak63b</filename>
	<releasename>Dark_Souls_Remastered_NSW-LiGHTFORCE</releasename>
	<trimmedsize>0</trimmedsize>
	<firmware></firmware>
	<type>1</type>
	<card></card>
</release>
```

I'm positive the bug should be fixed by now, but I want to make sure.

**Comment by Matt07211** [comment]
It didn't crash! Yay, I believe we can close this issue (I'll leave that up to you to do tho)

**Comment by DarkMatterCore** [comment]
Thanks! I'll create a new release right away.

**Comment by DarkMatterCore** [comment]
Fixed in fd4b7e933b57d76c53d8d931cb6a7b95ba4e11b7.
